xmppserver/src/test/java/org/jivesoftware/openfire/session/LocalIncomingServerSessionTest.java

```diff
@@ -106,43 +106,9 @@ public void setUpClass() throws Exception {
         sessionManager.initialize(xmppServer);
         doReturn(sessionManager).when(xmppServer).getSessionManager();
 
-        final ConnectionManager connectionManager = Fixtures.mockConnectionManager();
-        final ConnectionListener connectionListener = Fixtures.mockConnectionListener();
-        doAnswer(new ConnectionConfigurationAnswer(identityStoreConfig, trustStoreConfig)).when(connectionListener).generateConnectionConfiguration();
```

Comment:
I looked into your concern about removing `ConnectionConfigurationAnswer`. What I've seen is that `ConnectionConfigurationAnswer` was used to mock the call to `generateConnectionConfiguration()` on the `ConnectionListener`. 

With your new commit you are supplying the configuration previously defined in the `ConnectionConfigurationAnswer` directly to `ConnectionListener`, when `generateConnectionConfiguration()` is called it uses the config you passed in:

```
connectionListener = new ConnectionListener(ConnectionType.SOCKET_S2S,
                ConnectionSettings.Server.OLD_SSLPORT,
                ConnectionManagerImpl.DEFAULT_SERVER_SSL_PORT,
                ConnectionSettings.Server.ENABLE_OLD_SSLPORT,
                "xmpp.server.processing.threads",
                null,
                JiveGlobals.getProperty(ConnectionSettings.Server.TLS_POLICY),
                ConnectionSettings.Server.AUTH_PER_CLIENTCERT_POLICY,
                null,
                identityStore.getConfiguration(),
                trustStore.getConfiguration(),
                ConnectionSettings.Server.COMPRESSION_SETTINGS);
```

Not mocking the `ConnectionListener` makes sense to me in the context of an incoming server test.

Reply:
Yeah, you're probably right. I was mostly worried about 'loosing' the ability to configure TLS-specifics - but I guess we can do that to the properties (as is done with most other configuration in this same test).